File: src/main/java/com/gclasscn/base/domain/PageList.java
package com.gclasscn.base.domain;

import java.util.List;

public class PageList<T> {

	private Integer record;     // 总记录条数
	private Integer totalPages; // 总页数
	private Integer size;       // 页面显示数据条数
	private List<T> data;       // 展示数据

	public PageList(Integer record, Integer size, List<T> data) {
		this.record = record;
		this.size = size;
		this.data = data;
		this.totalPages = record % size == 0 ? record / size : record / size + 1;
	}

	public Integer getRecord() {
		return record;
	}

	public Integer getTotalPages() {
		return totalPages;
	}

	public Integer getSize() {
		return size;
	}

	public List<T> getData() {
		return data;
	}

}

File: src/main/java/com/gclasscn/base/domain/PageList.kt
package com.gclasscn.base.domain

import java.util.ArrayList

class PageList<T>(record: Int, size: Int, data: ArrayList<T>) {

	var record: Int = record     // 总记录条数
	var totalPages: Int? = null // 总页数
	var size: Int = size       // 页面显示数据条数
	var data: ArrayList<T> = data       // 展示数据

	init {
		totalPages = if(record % size == 0) record / size else record / size + 1
	}

}
